torch/_dynamo/source.py

```diff
@@ -258,3 +258,15 @@ def name(self):
 
     def make_guard(self, fn, is_volatile=False):
         raise NotImplementedError()
+
+
+# This is a synthetic source that is associated with the singleton
+# shape env guard we always register for all frames.  We get the actual
+# guard contents from the ambient ShapeEnv
+@dataclasses.dataclass
+class ShapeEnvSource(Source):
+    def name(self):
+        return ""
```

Comment:
Do we expect name to get called? I feel like this should maybe raise...

Reply:
The current convention, based on RandomValueSource, is that sources that don't have meaningful names return an empty string, and then we assert on them in guard builder time.

name IS called even for these sources, since we do not store Source directly in Guard, we only ever store the name. So you cannot (easily) make it raise.